js/ad_manager_vast.js

```diff
@@ -671,80 +770,101 @@ OO.Ads.manager(function(_, $) {
       // filter our playable stream:
-      var firstLinearAd = _.find(this.inlineAd.ads, function(v){ return !_.isEmpty(v.linear.mediaFiles); }, this);
-      if (!firstLinearAd) { return false; }
-      var streams = firstLinearAd.linear.mediaFiles;
-      var maxMedia = _.max(streams, function(v) { return parseInt(v.bitrate, 10); });
-      this.vastAdUnit.maxBitrateStream = maxMedia && maxMedia.url;
-      this.vastAdUnit.durationInMilliseconds = OO.timeStringToSeconds(firstLinearAd.linear.Duration) * 1000;
-      _.extend(this.vastAdUnit.data, firstLinearAd);
-      this.vastAdUnit.data.tracking = firstLinearAd.linear.tracking;
-      addToTimeline(this.vastAdUnit, adLoaded);
-      if (_.isEmpty(this.vastAdUnit.streams)) {
-        // No Playable stream, report error.
-        OO.log("Can not find playable stream in vast result", this.inlineAd);
+      if (_.isEmpty(ad.linear.mediaFiles)) {
         return false;
       }
-      return true;
+      params = params ? params : {};
+      var mediaFiles = ad.linear.mediaFiles;
+      var maxMedia = _.max(mediaFiles, function(v) { return parseInt(v.bitrate, 10); });
+      var vastAdUnit = { data: {}, vastUrl: this.vastUrl, maxBitrateStream: null };
+      vastAdUnit.maxBitrateStream = maxMedia && maxMedia.url;
+      vastAdUnit.durationInMilliseconds = OO.timeStringToSeconds(ad.linear.Duration) * 1000;
+      _.extend(vastAdUnit.data, ad);
+      vastAdUnit.data.tracking = ad.linear.tracking;
+      vastAdUnit.adPodIndex = params.adPodIndex ? params.adPodIndex : 1;
+      vastAdUnit.adPodLength = params.adPodLength ? params.adPodLength : 1;
+      vastAdUnit.positionSeconds = adLoaded.time/1000;
+
+      // Save the stream data for use by VideoController
+      var streams = {};
+      var linearAd = ad.linear;
+      if (linearAd && linearAd.mediaFiles) {
+        var vastStreams = linearAd.mediaFiles;
+        var videoEncodingsSupported = OO.VIDEO.ENCODING;
+        var streamData;
+        for (var encoding in videoEncodingsSupported) {
+          streamData = null;
+          streamData = this._extractStreamForType(vastStreams, videoEncodingsSupported[encoding]);
+          if (streamData) {
+            streams[videoEncodingsSupported[encoding]] = streamData;
+          }
+        }
+      }
+
+      vastAdUnit.streams = streams;
+      return vastAdUnit;
     };
 
     /**
      * If a non-linear Ad is found then it is parsed and added to the timeline via the addToTimeline function.
      * @public
      * @method Vast#_handleNonLinearAd
+     * @param {object} ad The ad object
      * @param {object} adLoaded The ad that was loaded
-     * @returns {boolean} True if the load was successful and a stream was found otherwise false.
+     * @param {object} params Additional params
+     * @returns {object} The ad unit object ready to be added to the timeline
      */
-    this._handleNonLinearAd = function(adLoaded) {
+    this._handleNonLinearAd = function(ad, adLoaded, params) {
```

Comment:
this function is public but is preceded by an underscore.  Is it meant to have been private?